ngraph/core/reference/src/runtime/reference/tile.cpp

```diff
@@ -14,34 +14,96 @@
 // limitations under the License.
 //*****************************************************************************
 
+#include <algorithm>
 #include <cmath>
-#include <stdio.h>
+#include <cstdio>
+#include <numeric>
 
 #include "ngraph/check.hpp"
 #include "ngraph/runtime/reference/tile.hpp"
 
 using namespace ngraph;
 
-void runtime::reference::tile(
-    const char* arg, char* out, const Shape& in_shape, const Shape& out_shape, size_t elem_size)
+namespace
+{
+    bool is_axis_upper_bound(const Shape& shape,
```

Comment:
Can you remove the `run` param from this function and check `axis-- == 0` in the main loop?

Reply:
Decrement of axis should happen in every iteration of inner while loop. If I move this check to the main loop it will decrease only once.

Reply:
Using struct that keep track of indices, axis and run may be the good solution. Additionally this structure can have is axis upper_bound method.